### PR TITLE
Change license to BSD-3-Clause

### PR DESCRIPTION
**Pull request by safl**
Apache 2 seems to not be compatible with GPL-v2, that is, a GPL-v2 licensed software cannot use a Apache 2 license software.
The license of xNVMe is intended to make xNVMe usable for any given project, this incompatibility with GPL-v2 restricts that.
Thus, this change from Apache 2.0 to the BSD 3-clause licence.

Needs review/confirmation from the following contributors:

- [x] Liad Oz <liadozil@gmail.com>
- [x] Vered Kalef <vered.kalef@gmail.com>

Hi @LiadOz and @vkalef , do you have any objections against changing license to BSD-3-Clause?

REUSE-Compliance
===============

While taking another pass on this then I stumbled on REUSE Software, for reference see

* https://reuse.software/
* https://fosdem.org/2023/schedule/event/sbom_reuse/
* https://github.com/fsfe/reuse-tool/blob/main/.pre-commit-config.yaml

This seems like a very well though-out approach handling licenses and copyright. Thus, since this PR touch most of everything when changing license, and it would great to **not** to that again. Then, it might be worth it, to ensure that xNVMe becomes REUSE-compliant.

- [x] Ensure license and copyright in files match REUSE-conventions
Seems like the following does the job:
```bash
reuse annotate  \
  --license BSD-3-Clause \
  --copyright "Samsung Electronics Co., Ltd" \
  -r \
  --copyright-style spdx \
  --exclude-year \
  examples tests tools lib include
```
- [x] Replace, the entire or parts of, ``third-party/README.rst`` with ``.reuse/dep5``
- [x] Add a pre-commit-hook checking REUSE compliance
```
repos:
  - repo: https://github.com/fsfe/reuse-tool
    rev: v1.1.2
    hooks:
      - id: reuse
```
- [x] Add a neat little REUSE-Software banner to README.rst
- [x] This will not be neat, until it reaches ``main``, could it check a different branch until then?

The advantage to taking the versioning-changes a step further with REUSE-compliance is that xNVMe will take a step to follow the gold-standard for open-source software and in practice having tools to assist maintaining REUSE-compliance. Thus when others look at the licences for xNVMe, then it will be very easy to get the information needed.

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# This file lists the official libvfn authors for copyright purposes. It is
```
Comment (Baekalfen) [inline AUTHORS:1]: > libvfn

Comment (safl) [reply AUTHORS]: thanks!

**Comment by LiadOz** [comment]
Looks good to me

**Comment by vkalef** [comment]
Hi
I approve.
regards,
Vered

**Comment by safl** [comment]
Thanks @LiadOz and @vkalef ! I will add your sign-off to the commit message then.

**Comment by vkalef** [comment]
Great, thanks!

**Pull request merged**